File: benchmarks/pandas/bench_series_round.py
"""Benchmark: series round (2 decimals) on 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = (np.arange(ROWS) % 10000) * 0.1234
s = pd.Series(data)

for _ in range(WARMUP):
    s.round(2)

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.round(2)
total = (time.perf_counter() - start) * 1000

print(json.dumps({ "function": "series_round", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))

File: benchmarks/tsb/bench_series_round.ts
/**
 * Benchmark: series round (2 decimals) on 100k-element Series
 */
import { Series, seriesRound } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Float64Array.from({ length: ROWS }, (_, i) => (i % 10000) * 0.1234);
const s = new Series(data);

for (let i = 0; i < WARMUP; i++) {
  seriesRound(s, { decimals: 2 });
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  seriesRound(s, { decimals: 2 });
}
const total = performance.now() - start;

console.log(JSON.stringify({ function: "series_round", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));
